### PR TITLE
Add new hook for dashboard.

### PR DESCRIPTION
**Pull request by DevelopmentCool2449**
I've noticed that using dashboard-after-initialize-hook with solaire doesn't fix banner image background issue.
I found that this is because the banner is created before solaire is called.

The solution i found is making other hook that just run before dashboard buffer is created.

I wonder if this is a can a good idea or just run dashboard-after-initialize-hook in dashboard--with-buffer.

Also i've noticed moving solaire first in dashboard-startupify-list fix this too.

### CODE REVIEW

File: dashboard.el [hunk 1]
```diff
@@ -440,6 +445,7 @@ Optional argument ARGS adviced function arguments."
   "Execute BODY in dashboard buffer."
   (declare (indent 0))
   `(with-current-buffer (get-buffer-create dashboard-buffer-name)
+     (run-hooks 'dashboard-before-initialize-hook)
```
Comment (jcs090218) [inline dashboard.el:448]: I don't think this should be inserted here.

File: dashboard.el [hunk 2]
```diff
@@ -86,6 +86,11 @@
   :group 'dashboard
   :type 'hook)
 
+(defcustom dashboard-before-initialize-hook nil
+  "Hook that is run before dashboard buffer is initialized."
+  :group 'dashboard
+  :type 'hook)
```
Comment (jcs090218) [inline dashboard.el:92]: Let’s put this above the after-initialize hook, so it makes a bit more sense.

Comment (DevelopmentCool2449) [reply dashboard.el]: fixed

File: dashboard.el (hunk 1 above)
Comment (DevelopmentCool2449) [reply dashboard.el:448]: should i move this into dashboard-insert-startupify-lists?

this hook must be run before dashboard inserts any widgets.

**Comment by DevelopmentCool2449** [comment]
I've moved before-initialize hook in insert-startupify-lists.

File: dashboard.el (hunk 1 above)
Comment (jcs090218) [reply dashboard.el:448]: Yes.

**Comment by DevelopmentCool2449** [comment]
I forgot delete dashboard-before-initialize-hook from dashboard--with-buffer.
I've fixed it.

**Review by jcs090218 — APPROVED** [review]

**Comment by jcs090218** [comment]
Thank you

**Pull request merged**